### PR TITLE
Changes needed to support importing as an ESM module

### PR DESCRIPTION
**Pull request by wlach**
* Change the type of the package
* Link to compiled distribution in package.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,14 +9,15 @@
   "module": "dist/esm/index.js",
   "unpkg": "dist/index.umd.min.js",
   "types": "dist/types/index.d.ts",
+  "type": "module",
```
Comment (wlach) [inline package.json:12]: This may have unwanted downstream impacts

**Comment by wlach** [comment]
Some small things I needed to do to make https://github.com/wlach/mystjs-svelte work. I'm still far from being a node expert, so I'm not 100% sure if the `"type": "module",` part is free of unwanted downstream consequences. Probably could use testing.

**Comment by rowanc1** [comment]
So cool. It may take a bit to test this out in all of the various places, are you able to move forward in your work without this being merged released in the next two weeks?

**Comment by wlach** [comment]
> So cool. It may take a bit to test this out in all of the various places, are you able to move forward in your work without this being merged released in the next two weeks?

Sorry for late reply. Yup, this is extreme prototype code so am not in any rush.

**Comment by rowanc1** [comment]
Hi @wlach, we are finally getting to some of these things. As part of the updates recently we removed the `exports` dict, which was causing some problems with the module loading. We also took some pieces of the module pointers which no longer point to the bundle.

I have not been able to get the `"type": "module"` working in downstream packages with out a ton of other changes.

I *think* that the current changes should be enough to improve your downstream experience (esm can consume cjs, but not the other way around) and I think that previously our ESM module was just wrong.

We are planning a new release in the coming days, and I would be curious if it fixes your use case! I will ping here when we have released.

**Comment by rowanc1** [comment]
Hi @wlach -- I am going to close this, as it is now quite out of date!

Happy to fix up install issues if you find any in the future. We have gotten better at our base scripts, so hopefully less problems next time. :)

**Pull request closed without merging**